### ISSUE TITLE
Some suggestions and feature requests

### ISSUE THREAD

**Issue by nik-gr** [Sav22999/websites-notes#46]
1. Can you pls make optional the permission for access data for all websites? I understand many people will like the feature of sticky notes but personally I'm not gonna use it, it doesn't really provide anything extra + I'm not sure if it is accessible by the webpage's developer. Also by having it as a required permission will create an obstacle to the number of users trying the addon... Many people avoid installing addons with such permissions unless they have been reviewed and characterized as 'Recommended' by firefox.

2. The only welcome addition from sticky notes is the resizable size. Will u consider to make the default note popup resizable too? Options I can think of: resizable global dimensions/by page/automatically adjusted to the note's text.

3. I asked you in the past about using * to add a note to more than one urls (#13 ). Sorry I didn't respond back then I had some issues with my github account. I'm not talking about some advanced pattern, just to be able to enter the * at the end of the selected part of url so it will match all with that prefix basically. There was one addon like that with the old ff and it used to be popular but as far as I know no addon nowdays provides that and I consider it a game changer. The main reason for a note addon is to be able to see the note if at sometime in the future u return back to the same url. There are websites that accept both x.y/z/ and x.y/z. If you revisit it (perhaps from a link) if it is not of the same form as when u wrote the note then u won't see the note. This is solved with * . Or there is something u want to keep in mind for a whole section of it but not for the whole domain, one easy example is qithub let's say I want to have a note regarding this repository https://github.com/Sav22999/websites-notes/*, or the developer, if I leave it on one of its pages whenever I visit some other I won't see it and if it is a repository I don't visit frequently I won't even remember it. And there are many other such cases: profiles in social networks where someone's profile has several different pages (\profilex\photos etc), same for tutorials or similar to github if your in the space of a specific item/software/.. in a portal that has many pages related to it, etc
(also in this subject if not difficult a choice to ignore the 1st part of url would be a welcome one as well = to display a note in all x.domain.com, y.domain.com and domain.com)
Probably that would be a solution for #19  as well 

4. Search: Can it be added a support to search multiple terms instead of one kewword/phrase as it is now (perhaps separated by comma)? And maybe adding the url in the searchable text?

5. If it is easy to implement, can you make the note html editable? I mean to have the ability to add html tags (for links, hilighting etc) - it can provide some kind of a workaround for #17  as well 

6. For non http/s urls there is no page option anymore, why? It is possible someone to want to add notes perhaps to  different pages of the same addon at moz-extension://... Without the page option he won't be able to do it. I understand it is a rare case (me, I don't think I will use it) but since the functionality is already implemented and exists, why to hide it

**Comment by Sav22999** [comment]
Hi! Thank you for all these suggestions. To be more useful I'll open single issues

1. Yes, I'll do this, thanks for the suggestion #47

2. I'll think about this #48

3. This would be interesting, although it would be complicated to manage. An example:
You save "Hello1" for "https://github.com/*", then you save "Hello2" for "https://github.com/Sav22999/*" and then you save "Hello3" for "https://github.com/Sav22999/websites-notes/*". Ok, now you to "https://github.com/Sav22999/websites-notes/issues", what it should show to you? Hello1, Hello2 or Hello3? And how you should call the tabs? I can't, right now, image a good UX/UI to show this (because, theoretically, it should show: "Hello1" and "Hello2" and "Hello3" #49

4. Absolutely, you're right and I'll add this! #50

5. There is already an issue dedicated to this. I don't think, anyway, I'll add that feature... not now, at least, sorry #17 

6. Ok, I'll see this and I'll re-add "Page" too. I thought it was useless, but it doesn't look so. Sorry #51

Thank you very much for all these precious feedback!

**Comment by nik-gr** [comment]
Hi, thank you for the addon and for being active and also for considering those suggestions!

As for 5, what I had in mind is a little different (that could  be somekind of a workaround for images as well)
Let's say that the text of the note includes html tags then when the popup opens or the global/domain/page label is clicked the note would be displayed after interpreting the html tags. Like that for 'normal' notes as they are now, the display will continue to be the same with the addition if there are html tags the note will have a rich display. And if the user gain focus in the note field so to modify the note (by clicking at the area anywhere that has no html event behavior - e.g. not at links) the editable format would show the pure html syntax ready to be modified.
The main usage I would had with this would be to add links related to the page/website etc that I can easily visit from the note just with a click on the `<a>` tag.
It's an addition, a welcome one, but not really so important related to the main, core functionalities that a note addon shall have. I don't even know if it is easy to implement. if it's difficult or requires big changes because of how the addon is designed now or if perhaps there are security concerns with such a functionality then I guess it's not worth to bother with it.

Again, thank you for considering them!

**Comment by nik-gr** [comment]
Yes, more or less that's what I had in mind.
I agree most users most likely won't use it, such an implementation shouldn't be a priority. I guess it's more about how much work is needed to do it and if u think it's worth it overall.
In any case thanks for considering it :)

**Comment by Sav22999** [comment]
@nik-gr in the 3.5 release you can now find:
- Improved searching
- Added "page" also for moz-extension (it's complicated add for "about" and others one)
- Optional permissions
- Other improvements

I'm working on also for "sub-domains" but it's complicated to implement a decent UX... don't worry, I'm working to it as well, so it should arrive in future

I hope you'll like new features and improvements 😄

**Comment by Cade66** [comment]
Notefox 3.7 introduced two new features: Subdomains (Advanced managing) and Notes' HTML tags.
Subdomains (Advanced managing) has an on/off option in Settings, unfortunately HTML tags does not.
I'm not fond of the HTML tags feature given Notefox is a note taker, not a text editor, but I understand some may be fond of html tags to fine-tune their notes' appearance.
Given HTML tags has no option in Settings to be turned off, and for those as myself who may wish to hide the feature, this little userContent.css will do the job.

```
@-moz-document url("moz-extension://[INTERNAL_UUID]/popup.html")
{
/* Remove html tags (format section) */
#format-buttons {display: none !important;}
#notes-section > #notes, textarea {margin-bottom:0px !important;}
}
```
This said and done, I emphasize on the new Subdomains (Advanced managing) feature which is IMO a true, worthy improvement.
Thnaks for this nice extension, best in its category IMO :)

**Comment by Sav22999** [comment]
@Cade66 so you mean to disable "buttons" to insert text formatting?

I'll add an option in the next release, thank you very much for your suggestion!

**Comment by Cade66** [comment]
@Sav22999 , indeed the option to disable it, only. Notefox settings as Notefox itself is nicely elaborated, adding the option to toggle text formatting is just a modest suggestion :)

**Comment by Sav22999** [comment]
@Cade66 Just released the 3.7.4 where you can find an option to disable the panel to use HTML tags.
Unluckily for now I don't know how to enable word-wrap in "pre" tags (which is pre-formatted).

I'll see something to add this feature although 😄

**Comment by Sav22999** [comment]
@Cade66 3.7.5 released! Now the text is word-wrap by default, but you can disable this in Settings 😄

**Comment by Cade66** [comment]
@Sav22999 , I don't know how you manage to add, modify, fix so quickly :)
As soon as 3.7.5 will be available on AMO I'll update.
Thanks for being committed to users' suggestions & wishes. I really appreciate that.

**Comment by Sav22999** [comment]
@Cade66 Haha I do this as hobby for now, so I dedicated all my free time and all my passion. Thanks to you because use the addon and suggest me always interesting features!

It should be already available on AMO. Let me know 😉

**Comment by Cade66** [comment]
@Sav22999 , I just updated to 3.7.5 : perfect :)
Added settings **_Show HTML text-formatting panel in the popup_** and **_Disable word-wrap notes text_** are flawless and bring the cherry on the cake!
I do hope other users will enjoy as I do. Nice work, thanks :)

**Comment by nik-gr** [comment]
thank you Saverio! 🙏

I also find word wrapping more friendly (and with implementation of #48 I don't think there will be any need for not using word wrapping in all cases). 
It may be a better choice to have it as default with possibility to change to no wrapping from settings, users who already have notes with long lines perhaps will find it strange to see all of a sudden their notes disheartening in a scrolling bar (just a thought)

I didn't understand what the last button of 'text decoration' does can you pls explain? Also how (or if) it is possible to add my own html tags
(btw key shortcuts I'm not sure if they are of any assistance since default ff actions exist for them)

**Comment by Sav22999** [comment]
@nik-gr Yes, it's set by default to "false" (the "Disable word-wrap"), so it's by default the text is word-wrapped 😄 (I've implemented that in this way for the same reason you explained!).

The last button is for spellcheck detection (you can disable). I'll add title on buttons to be more clear.
Actually that's not the "default" shortcuts... I've implemented manually 🤣  but some people, maybe, need buttons... so, I think to have buttons as "option" (to disable in case) it's the best solution 😄 

The spellcheck detection, for now, didn't save your choice and you need to change each time you open the popup... I'll add an option for this as well!

**Comment by Sav22999** [comment]
@nik-gr No, actually the abilty to paste images remains. The option in Settings **only** disable the panel with buttons, but you can insert images and text-formatting via shortcuts 😄 

I'll see about the "link" feature, but it could be unsure for some users I think

**Comment by nik-gr** [comment]
Didn't think of that possibility. I guess a difference on how a user and a developer sees an app :)
I'm not sure if there is real risk from something like that, but if there is yes I would say it's better to avoid adding it (from developer's perspective) - and although a welcome addition, it is not a really important one so it won't matter much if it's not gonna happen

**Comment by Sav22999** [comment]
@nik-gr Hi, and thanks!

About the shortcuts: unfortunately every user has own keyboard shortcuts. The "link" one it's like a "hidden feature" now, because in feature I should improve it (I'll add a button like "bold", "italic" etc. where you should be able to choose the link as well, maybe.
Maybe I can make this editable in Settings in future... I'll see

About the tab position... actually I don't think I can choose it: you should choose this option in Firefox settings (I think), I can only "open" a new tab, not choosing the position, but I'll check better 😄 

If you use a Mac you can use the shortcut ("Option"+Key, instead of "Command"+Key)

Thank you very much for your suggestions!

**Comment by nik-gr** [comment]
Ok it's strange because it's the normal behavior when I open a url from a link to open next to current but you are right there is an option, when browser.tabs.insertAfterCurrent = true  it does what I asked for the links in notes as well while not changing something else so no other action needed 👍

About the shortcut iit's not important it doesn't break any functionality, just a little inconvenient. I wrote about it because it's an interfere with a default firefox action. But I can see your point, if you hardcode it to something else perhaps it will interfere with other users own shortcuts in a more serious way. Perhaps it will be a good option if you eventually make it editable

**Comment by nik-gr** [comment]
> here is an option, when browser.tabs.insertAfterCurrent = true it does what I asked for the links in notes as well while not changing something else

Actually it changes where a new tab with ctrl-t will open which now opens next to the active one and that is not a desired behavior. Since I come across that much more often than the need to open a link from a note I will change that preference value back to its default. 
But there must be a command that will allow the addon to open a url in a tab next to the active one, I have other addon which does that. Definetely it's not a priority but if u can impllement it when you'll have the time it will be a welcome addition

**Comment by nik-gr** [comment]
🙏

**Comment by Sav22999** [comment]
Implemented in the version 4.6
https://www.notefox.eu/help/notefox-4.6/
